diff -Nur aterm-1.0.1/autoconf/config.guess aterm-1.0.1-irix/autoconf/config.guess
--- aterm-1.0.1/autoconf/config.guess	2004-12-06 20:27:29.000000000 +0400
+++ aterm-1.0.1-irix/autoconf/config.guess	2025-08-10 16:39:04.133151350 +0400
@@ -914,6 +914,9 @@
     sparc:Linux:*:* | sparc64:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit 0 ;;
+    aarch64:Linux:*:*)
+	echo aarch64-unknown-linux-gnu
+	exit 0 ;;
     x86_64:Linux:*:*)
 	echo x86_64-unknown-linux-gnu
 	exit 0 ;;
diff -Nur aterm-1.0.1/autoconf/configure.in aterm-1.0.1-irix/autoconf/configure.in
--- aterm-1.0.1/autoconf/configure.in	2007-08-01 22:45:34.000000000 +0500
+++ aterm-1.0.1-irix/autoconf/configure.in	2025-08-10 16:38:03.981379205 +0400
@@ -207,10 +207,6 @@
   CFLAGS="$CFLAGS -Wuninitialized -Wwrite-strings -Wcast-qual -Wbad-function-cast -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wnested-externs -Wconversion -Wcomment -Wcast-align -Winline -Wshadow -Wredundant-decls -Wid-clash-31"
 fi
 
-if test "x$GCC" = "xyes"; then
-   LDFLAGS="$LDFLAGS -rdynamic"
-fi
-
 dnl# done with flags - lets export those for our children to use :
 export CFLAGS
 export LDFLAGS
diff -Nur aterm-1.0.1/autoconf/Make.common.in aterm-1.0.1-irix/autoconf/Make.common.in
--- aterm-1.0.1/autoconf/Make.common.in	2005-05-26 21:44:49.000000000 +0500
+++ aterm-1.0.1-irix/autoconf/Make.common.in	2025-08-10 16:38:03.981551698 +0400
@@ -41,7 +41,7 @@
 CP = @CP@
 SED = @SED@
 INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL@ -s -m 755
+INSTALL_PROGRAM = @INSTALL@ -m 755
 INSTALL_DATA = @INSTALL@ -m 644
 
 # Flags & libs
diff -Nur aterm-1.0.1/configure aterm-1.0.1-irix/configure
--- aterm-1.0.1/configure	2007-08-01 22:47:04.000000000 +0500
+++ aterm-1.0.1-irix/configure	2025-08-10 16:38:03.982361001 +0400
@@ -4571,10 +4571,6 @@
   CFLAGS="$CFLAGS -Wuninitialized -Wwrite-strings -Wcast-qual -Wbad-function-cast -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wnested-externs -Wconversion -Wcomment -Wcast-align -Winline -Wshadow -Wredundant-decls -Wid-clash-31"
 fi
 
-if test "x$GCC" = "xyes"; then
-   LDFLAGS="$LDFLAGS -rdynamic"
-fi
-
 export CFLAGS
 export LDFLAGS
 
diff -Nur aterm-1.0.1/doc/Makefile.in aterm-1.0.1-irix/doc/Makefile.in
--- aterm-1.0.1/doc/Makefile.in	2005-01-11 23:05:17.000000000 +0400
+++ aterm-1.0.1-irix/doc/Makefile.in	2025-08-10 16:38:03.982937771 +0400
@@ -29,7 +29,7 @@
 	$(RMF) Makefile
 
 install:
-	$(INSTALL_DATA) aterm.1 $(DESTDIR)$(mandir)/$(manprefix)aterm.$(manext)
+	$(INSTALL_DATA) aterm.1 $(DESTDIR)$(mandir)/$(manprefix)aterm.$(manext) || true
 
 uninstall:
 	-cd $(mandir); $(RMF) $(manprefix)aterm.$(manext)
