diff -Nur bash-5.2.37/config-bot.h bash-5.2.37-irix/config-bot.h
--- bash-5.2.37/config-bot.h	2021-11-05 13:11:04.000000000 +0000
+++ bash-5.2.37-irix/config-bot.h	2025-05-23 11:37:39.253161369 +0000
@@ -205,3 +205,6 @@
 
 /* If you don't want bash to provide a default mail file to check. */
 /* #undef DEFAULT_MAIL_DIRECTORY */
+
+
+#include <math.h>
diff -Nur bash-5.2.37/configure bash-5.2.37-irix/configure
--- bash-5.2.37/configure	2024-09-23 21:23:59.000000000 +0000
+++ bash-5.2.37-irix/configure	2025-05-23 11:37:39.252161311 +0000
@@ -16289,7 +16289,7 @@
 ac_fn_c_check_func "$LINENO" "mbscasecmp" "ac_cv_func_mbscasecmp"
 if test "x$ac_cv_func_mbscasecmp" = xyes
 then :
-  printf "%s\n" "#define HAVE_MBSCMP 1" >>confdefs.h
+  printf "%s\n" "#define HAVE_MBSCASECMP 1" >>confdefs.h
 
 fi
 
@@ -18720,7 +18720,7 @@
 
 	fi
 if test $bash_cv_type_socklen_t = no; then
-  printf "%s\n" "#define socklen_t unsigned int" >>confdefs.h
+  printf "%s\n" "#define socklen_t uint32_t" >>confdefs.h
 
 fi
 
diff -Nur bash-5.2.37/examples/loadables/finfo.c bash-5.2.37-irix/examples/loadables/finfo.c
--- bash-5.2.37/examples/loadables/finfo.c	2020-06-29 14:56:32.000000000 +0000
+++ bash-5.2.37-irix/examples/loadables/finfo.c	2025-05-23 11:37:39.252161311 +0000
@@ -34,6 +34,7 @@
 #ifdef MAJOR_IN_SYSMACROS
 #  include <sys/sysmacros.h>
 #endif
+#include <sys/mkdev.h>
 #include "posixstat.h"
 #include <stdio.h>
 #include <pwd.h>
diff -Nur bash-5.2.37/lib/readline/rlprivate.h bash-5.2.37-irix/lib/readline/rlprivate.h
--- bash-5.2.37/lib/readline/rlprivate.h	2022-08-11 22:35:16.000000000 +0000
+++ bash-5.2.37-irix/lib/readline/rlprivate.h	2025-05-23 11:40:12.737968144 +0000
@@ -23,6 +23,10 @@
 #if !defined (_RL_PRIVATE_H_)
 #define _RL_PRIVATE_H_
 
+#ifdef sgi
+#include <sys/signal.h>
+#endif
+
 #include "rlconf.h"	/* for VISIBLE_STATS */
 #include "rlstdc.h"
 #include "posixjmp.h"	/* defines procenv_t */
diff -Nur bash-5.2.37/Makefile.in bash-5.2.37-irix/Makefile.in
--- bash-5.2.37/Makefile.in	2022-07-26 20:24:54.000000000 +0000
+++ bash-5.2.37-irix/Makefile.in	2025-05-23 11:37:39.250161195 +0000
@@ -846,7 +846,6 @@
 		infodir=$(infodir) htmldir=$(htmldir) DESTDIR=$(DESTDIR) $@ )
 	-( cd $(DEFDIR) ; $(MAKE) $(MFLAGS) DESTDIR=$(DESTDIR) $@ )
 	-( cd $(PO_DIR) ; $(MAKE) $(MFLAGS) DESTDIR=$(DESTDIR) $@ )
-	-( cd $(LOADABLES_DIR) && $(MAKE) $(MFLAGS) DESTDIR=$(DESTDIR) $@ )
 
 install-strip:
 	$(MAKE) $(MFLAGS) INSTALL_PROGRAM='$(INSTALL_PROGRAM) -s' \
diff -Nur bash-5.2.37/parse.y bash-5.2.37-irix/parse.y
--- bash-5.2.37/parse.y	2024-09-23 21:24:01.000000000 +0000
+++ bash-5.2.37-irix/parse.y	2025-05-23 11:37:39.252161311 +0000
@@ -2640,6 +2640,7 @@
       parser_state |= PST_ENDALIAS;
       /* We need to do this to make sure last_shell_getc_is_singlebyte returns
 	 true, since we are returning a single-byte space. */
+#if defined (HANDLE_MULTIBYTE)
       if (shell_input_line_index == shell_input_line_len && last_shell_getc_is_singlebyte == 0)
 	{
 #if 0
@@ -2653,6 +2654,7 @@
 	  shell_input_line_property[shell_input_line_index - 1] = 1;
 #endif
 	}
+#endif /* HANDLE_MULTIBYTE */
       return ' ';	/* END_ALIAS */
     }
 #endif
diff -Nur bash-5.2.37/y.tab.c bash-5.2.37-irix/y.tab.c
--- bash-5.2.37/y.tab.c	2024-09-23 21:24:01.000000000 +0000
+++ bash-5.2.37-irix/y.tab.c	2025-05-23 11:37:39.253161369 +0000
@@ -4955,6 +4955,7 @@
       parser_state |= PST_ENDALIAS;
       /* We need to do this to make sure last_shell_getc_is_singlebyte returns
 	 true, since we are returning a single-byte space. */
+#if defined (HANDLE_MULTIBYTE)
       if (shell_input_line_index == shell_input_line_len && last_shell_getc_is_singlebyte == 0)
 	{
 #if 0
@@ -4968,6 +4969,7 @@
 	  shell_input_line_property[shell_input_line_index - 1] = 1;
 #endif
 	}
+#endif /* HANDLE_MULTIBYTE */
       return ' ';	/* END_ALIAS */
     }
 #endif
