diff -Nur gettext-0.24/gettext-tools/gnulib-lib/csharpcomp.c gettext-0.24-irix/gettext-tools/gnulib-lib/csharpcomp.c
--- gettext-0.24/gettext-tools/gnulib-lib/csharpcomp.c	2025-02-25 09:13:39.000000000 +0000
+++ gettext-0.24-irix/gettext-tools/gnulib-lib/csharpcomp.c	2025-04-26 10:42:21.000000000 +0000
@@ -63,7 +63,7 @@
  */
 
 static int
-name_is_dll (const struct dirent *d)
+name_is_dll (struct dirent *d)
 {
   if (d->d_name[0] != '.')
     {
diff -Nur gettext-0.24/gettext-tools/gnulib-lib/csharpexec.c gettext-0.24-irix/gettext-tools/gnulib-lib/csharpexec.c
--- gettext-0.24/gettext-tools/gnulib-lib/csharpexec.c	2025-01-02 00:27:37.000000000 +0000
+++ gettext-0.24-irix/gettext-tools/gnulib-lib/csharpexec.c	2025-04-26 10:41:39.000000000 +0000
@@ -100,7 +100,7 @@
  */
 
 static int
-name_is_dll (const struct dirent *d)
+name_is_dll (struct dirent *d)
 {
   if (d->d_name[0] != '.')
     {
diff -Nur gettext-0.24/gettext-tools/gnulib-tests/select.c gettext-0.24-irix/gettext-tools/gnulib-tests/select.c
--- gettext-0.24/gettext-tools/gnulib-tests/select.c	2025-01-02 00:27:37.000000000 +0000
+++ gettext-0.24-irix/gettext-tools/gnulib-tests/select.c	2025-04-26 13:23:22.791847673 +0000
@@ -566,6 +566,8 @@
 
 #undef select
 
+extern int select(int, fd_set *, fd_set *, fd_set *, struct timeval *);
+
 int
 rpl_select (int nfds, fd_set *rfds, fd_set *wfds, fd_set *xfds,
             struct timeval *timeout)
