diff -Nur gnutls-3.7.11/lib/x509/hostname-verify.c gnutls-3.7.11-irix/lib/x509/hostname-verify.c
--- gnutls-3.7.11/lib/x509/hostname-verify.c	2024-05-15 12:10:27.000000000 +0000
+++ gnutls-3.7.11-irix/lib/x509/hostname-verify.c	2025-04-29 07:29:01.816870117 +0000
@@ -168,6 +168,7 @@
 	    ((p=strchr(hostname, ':')) != NULL || inet_pton(AF_INET, hostname, &ipv4) != 0)) {
 
 		if (p != NULL) {
+#ifdef HAVE_IPV6
 			struct in6_addr ipv6;
 
 			ret = inet_pton(AF_INET6, hostname, &ipv6);
@@ -176,6 +177,9 @@
 				goto hostname_fallback;
 			}
 			ret = check_ip(cert, &ipv6, 16);
+#else
+			return 0;
+#endif
 		} else {
 			ret = check_ip(cert, &ipv4, 4);
 		}
diff -Nur gnutls-3.7.11/src/gl/select.c gnutls-3.7.11-irix/src/gl/select.c
--- gnutls-3.7.11/src/gl/select.c	2024-05-23 12:25:58.000000000 +0000
+++ gnutls-3.7.11-irix/src/gl/select.c	2025-04-29 09:16:29.169859749 +0000
@@ -560,7 +560,9 @@
 #include <errno.h>
 #include <unistd.h>
 
+#ifndef sgi
 #undef select
+#endif
 
 int
 rpl_select (int nfds, fd_set *rfds, fd_set *wfds, fd_set *xfds,
