diff -Nur mc-4.8.0/lib/strutil/strutil8bit.c mc-4.8.0-irix/lib/strutil/strutil8bit.c
--- mc-4.8.0/lib/strutil/strutil8bit.c	2011-10-18 11:39:19.000000000 +0000
+++ mc-4.8.0-irix/lib/strutil/strutil8bit.c	2025-04-28 12:09:48.791390609 +0000
@@ -34,6 +34,10 @@
 #include "lib/global.h"
 #include "lib/strutil.h"
 
+#ifndef isblank
+#define isblank(c) ((c) == ' ' || (c) == '\t')
+#endif
+
 /* functions for singlebyte encodings, all characters have width 1
  * using standard system functions
  * there are only small differences between functions in strutil8bit.c 
diff -Nur mc-4.8.0/lib/tty/tty-ncurses.c mc-4.8.0-irix/lib/tty/tty-ncurses.c
--- mc-4.8.0/lib/tty/tty-ncurses.c	2011-10-18 11:39:19.000000000 +0000
+++ mc-4.8.0-irix/lib/tty/tty-ncurses.c	2025-04-28 12:30:39.819855676 +0000
@@ -181,11 +181,21 @@
     ESCDELAY = 200;
 #endif /* HAVE_ESCDELAY */
 
+    struct termios tty;
+    tcgetattr(STDIN_FILENO, &tty);  // Get current terminal settings
+
+    tty.c_cc[VINTR] = CTRL ('g');   /* ^g */
+    /* disable SIGQUIT to allow use Ctrl-\ key */
+    tty.c_cc[VQUIT] = NULL_VALUE;
+    tcsetattr (STDIN_FILENO, TCSANOW, &tty);
+
+#if 0
     /* use Ctrl-g to generate SIGINT */
     cur_term->Nttyb.c_cc[VINTR] = CTRL ('g');   /* ^g */
     /* disable SIGQUIT to allow use Ctrl-\ key */
     cur_term->Nttyb.c_cc[VQUIT] = NULL_VALUE;
-    tcsetattr (cur_term->Filedes, TCSANOW, &cur_term->Nttyb);
+    tcsetattr (cur_term->Filedes, TCSANOW, (const struct termios *)&cur_term->Nttyb);
+#endif
 
     tty_start_interrupt_key ();
 
diff -Nur mc-4.8.0/lib/vfs/parse_ls_vga.c mc-4.8.0-irix/lib/vfs/parse_ls_vga.c
--- mc-4.8.0/lib/vfs/parse_ls_vga.c	2011-10-18 11:39:19.000000000 +0000
+++ mc-4.8.0-irix/lib/vfs/parse_ls_vga.c	2025-04-28 12:09:48.792390629 +0000
@@ -40,6 +40,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+#include <sys/types.h>
+#include <sys/mkdev.h>
+
 #include "lib/global.h"
 #include "lib/widget.h"         /* message() */
 
diff -Nur mc-4.8.0/src/vfs/ftpfs/ftpfs.c mc-4.8.0-irix/src/vfs/ftpfs/ftpfs.c
--- mc-4.8.0/src/vfs/ftpfs/ftpfs.c	2011-10-19 11:30:59.000000000 +0000
+++ mc-4.8.0-irix/src/vfs/ftpfs/ftpfs.c	2025-04-28 12:09:48.793390650 +0000
@@ -187,6 +189,8 @@
 typedef int socklen_t;
 #endif
 
+#include <netdb.h>
+
 /* This should match the keywords[] array below */
 typedef enum
 {
@@ -1135,9 +1139,9 @@
     case AF_INET:
         ((struct sockaddr_in *) sa)->sin_port = port;
         break;
-    case AF_INET6:
-        ((struct sockaddr_in6 *) sa)->sin6_port = port;
-        break;
+//    case AF_INET6:
+//        ((struct sockaddr_in6 *) sa)->sin6_port = port;
+//        break;
     }
 
     return (connect (my_socket, (struct sockaddr *) sa, *salen) < 0) ? 0 : 1;
@@ -1185,10 +1189,10 @@
         af = FTP_INET;
         port = ((struct sockaddr_in *) &data_addr)->sin_port;
         break;
-    case AF_INET6:
-        af = FTP_INET6;
-        port = ((struct sockaddr_in6 *) &data_addr)->sin6_port;
-        break;
+//    case AF_INET6:
+//        af = FTP_INET6;
+//        port = ((struct sockaddr_in6 *) &data_addr)->sin6_port;
+//        break;
         /* Not implemented */
     default:
         return 0;
@@ -1265,9 +1269,9 @@
     case AF_INET:
         ((struct sockaddr_in *) data_addr)->sin_port = 0;
         break;
-    case AF_INET6:
-        ((struct sockaddr_in6 *) data_addr)->sin6_port = 0;
-        break;
+//    case AF_INET6:
+//        ((struct sockaddr_in6 *) data_addr)->sin6_port = 0;
+//        break;
     default:
         vfs_print_message (_("ftpfs: invalid address family"));
         ERRNOR (EINVAL, -1);
