diff -Nur readline-8.2/input.c readline-8.2-irix/input.c
--- readline-8.2/input.c	2022-04-08 19:43:24.000000000 +0000
+++ readline-8.2-irix/input.c	2025-04-28 08:13:51.596515130 +0000
@@ -37,6 +37,8 @@
 #  include <sys/file.h>
 #endif /* HAVE_SYS_FILE_H */
 
+#include <sys/select.h>
+
 #if defined (HAVE_UNISTD_H)
 #  include <unistd.h>
 #endif /* HAVE_UNISTD_H */
@@ -807,8 +809,8 @@
   int fd;
 #if defined (HAVE_PSELECT)
   sigset_t empty_set;
-  fd_set readfds;
 #endif
+  fd_set readfds;
 
   fd = fileno (stream);
   while (1)
