diff -Nur wget-1.25.0/lib/select.c wget-1.25.0-irix/lib/select.c
--- wget-1.25.0/lib/select.c	2024-11-10 21:27:18.000000000 +0000
+++ wget-1.25.0-irix/lib/select.c	2025-04-29 12:19:21.359401926 +0000
@@ -564,7 +564,9 @@
 #include <errno.h>
 #include <unistd.h>
 
+#ifndef sgi
 #undef select
+#endif
 
 int
 rpl_select (int nfds, fd_set *rfds, fd_set *wfds, fd_set *xfds,
diff -Nur wget-1.25.0/lib/vasnprintf.c wget-1.25.0-irix/lib/vasnprintf.c
--- wget-1.25.0/lib/vasnprintf.c	2024-11-10 21:27:18.000000000 +0000
+++ wget-1.25.0-irix/lib/vasnprintf.c	2025-04-29 12:19:21.360401977 +0000
@@ -211,11 +211,9 @@
    /* Unix.  */
 #  define SNPRINTF snprintf
    /* Here we need to call the native snprintf, not rpl_snprintf.  */
-#  undef snprintf
 # endif
 #endif
 /* Here we need to call the native sprintf, not rpl_sprintf.  */
-#undef sprintf
 
 /* GCC >= 4.0 with -Wall emits unjustified "... may be used uninitialized"
    warnings in this file.  Use -Dlint to suppress them.  */
diff -Nur wget-1.25.0/src/init.c wget-1.25.0-irix/src/init.c
--- wget-1.25.0/src/init.c	2024-11-10 14:44:08.000000000 +0000
+++ wget-1.25.0-irix/src/init.c	2025-04-29 12:19:21.360401977 +0000
@@ -922,7 +922,7 @@
   assert (0 <= comind && ((size_t) comind) < countof (commands));
 
   if ((unsigned) comind >= countof (commands))
-    return NULL;
+    return 0;
 
   DEBUGP (("Setting %s (%s) to %s\n", com, commands[comind].name, val));
   return commands[comind].action (com, val, commands[comind].place);
diff -Nur wget-1.25.0/src/utils.c wget-1.25.0-irix/src/utils.c
--- wget-1.25.0/src/utils.c	2024-11-10 17:46:46.000000000 +0000
+++ wget-1.25.0-irix/src/utils.c	2025-04-29 12:21:20.796461675 +0000
@@ -51,6 +51,8 @@
 
 #include <sys/stat.h>
 
+#include <math.h>
+
 /* For TIOCGWINSZ and friends: */
 #ifndef WINDOWS
 # include <sys/ioctl.h>
